pkg/interface/chat/src/js/components/lib/message.js

```diff
@@ -54,47 +56,74 @@ export class Message extends Component {
       );
     } else if ('me' in letter) {
       return (
-        <p className='body-regular-400 v-top'>
+        <p className='f7 lh-copy v-top'>
           {letter.me}
         </p>
       );
     } else {
-      return (
-        <p className='body-regular-400 v-top'>
-          {letter.text}
-        </p>
-      );
-    }
+      let chatroom = letter.text.match(
+        /(~[a-z]{3,6})(-[a-z]{6})?([/])(([a-z])+([/-])?)+/
+        );
+      if ((chatroom !== null) // matched possible chatroom
+        && (chatroom[1].length > 2) // possible ship?
+        && (urbitOb.isValidPatp(chatroom[1]) // valid patp?
+        && (chatroom[0] === letter.text))) { // entire message is room name?
```

Comment:
oof. good, but unwieldy. we really need a better way to send around links in an ongoing fashion.

Reply:
regexing a message for valid room names that can now have any number of hyphens and slashes without catching stack traces and URLs is surprisingly annoying.